### PR TITLE
Feat/ expense account

### PR DESCRIPTION
**Pull request by aliassan**
# Description

Expense account smart contract implementation.

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# Checklist:

* **Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by aliassan** [comment]
That's how it's working right now, it's deployed or created by the team owner, they deposit funds and they approve users to access. Based on the unit tests I was running in the afternoon all the possible budget limits are already working as well including the simple amount per transaction.

**Comment by hermannleboss** [comment]
Yes but doing it with all this sign-in stuff make the Smart Contract More complicated and will make the user sepend more in gas fee. 

Can you make it simple, with simple variable, simple function like: 
- uint256 maxLimit;
- function setMaxLimit()

Simple like that, easy to implement, easy to test, easy to go.

**Comment by aliassan** [comment]
> Yes but doing it with all this sign-in stuff make the Smart Contract More complicated and will make the user sepend more in gas fee.
> 
> Can you make it simple, with simple variable, simple function like:
> 
> * uint256 maxLimit;
> * function setMaxLimit()
> 
> Simple like that, easy to implement, easy to test, easy to go.

It's okay let me redo

File: contract/contracts/beacons/FactoryBeacon.sol [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./UserBeaconProxy.sol";
+
+contract FactoryBeacon is UpgradeableBeacon {
+  event BeaconProxyCreated(address indexed proxy, address indexed deployer);
+
+  constructor(address implementation) UpgradeableBeacon(implementation, msg.sender) {}
+
```
Comment (hermannleboss) [inline contract/contracts/beacons/FactoryBeacon.sol:12]: I have a problem from this line, but i think it's coming from the model @farrellh1 create. 

The implementation name should be implementationAddress so we no what type of value we are using.

**Review by hermannleboss — APPROVED** [review]
LGTM

**Pull request merged**